File: Euler/no2.js
'use strict';

function fib(a,b,sum) {
	var c = a + b,
			sum = sum
	console.log(c)
	if (c % 2 == 0) sum += c
	if (c < 4000000) 
		fib(b,c,sum)
	else
		console.log('sum: ' + sum)
}

fib(0,1,0)

// 4613732

File: Euler/no2.ex
# Euler Problem 2, "Even Fibonacci Numbers"

# recursively sum the even Fibonacci numbers
# smaller than 4,000,000

defmodule Fibonacci do
	def sumEven(a,b,sum) do
		c = a + b
		IO.puts c
		case c do
			c when (rem(c,2) == 0) and (c < 4000000) ->
				here = sum + c
				sumEven(b,c,here)
			c when (rem(c,2) !== 0) and (c < 4000000) ->
				sumEven(b,c,sum)
			c when (c > 4000000) ->
				IO.puts "sum: #{sum}"
		end
	end
end

Fibonacci.sumEven(0,1,0)

# 4613732